Adding documentation to Gamepad vibrationActuator and playEffect (#11967)
mdn/content: files/en-us/web/api/gamepadhapticactuator/playeffect/index.md

---
title: GamepadHapticActuator.playEffect()
slug: Web/API/GamepadHapticActuator/playEffect
tags:
  - API
  - Experimental
  - Gamepad
  - Gamepad API
  - GamepadHapticActuator
  - Method
  - Reference
  - playEffect
browser-compat: api.GamepadHapticActuator.playEffect
---
{{APIRef("Gamepad")}}{{SeeCompatTable}}

The **`playEffect()`** method of the {{domxref("GamepadHapticActuator")}} interface makes the hardware play a specific vibration pattern.

## Syntax

```js

> gamepadHapticActuatorInstance.playEffect(type, params).then(function(result) { /* ... */ });

Reviewer: ```suggestion
playEffect(type, params);
```

Reviewer: https://developer.mozilla.org/en-US/docs/MDN/Structures/Syntax_sections#concise_syntax_blocks

```

### Parameters

- _type_

>   - : A string representing the desired effect. This can vary depending on the hardware type, possible values are "dual-rumble" or "vibration".

Reviewer: ```suggestion
  - : A string representing the desired effect. This can vary depending on the hardware type. Possible values are "dual-rumble" or "vibration".
```

Reviewer: >This can vary depending on the hardware type.

Are you saying that either vibration or dual-rumble aren't available on some devices?

Author: My understanding is:

A Gamepad may not have a haptic actuator. If the gamepad has a haptic actuator, it will be either a "vibration" or a "dual-rumble" type.

The type of the haptic actuator defines what kind of effects are supported. The possible effects may be interchangeable in the future, regardless of actuator type, but that's not specified in the spec today.

The [excerpts](https://docs.google.com/document/d/1jPKzVRNzzU4dUsvLpSXm1VXPQZ8FP-0lKMT-R_p-s6g/edit#) that led me to this interpretation:

> _Two types of haptic actuators are defined. A "vibration" actuator is a haptic feedback device that supports vibration. This type of actuator is appropriate for gamepads with a single vibration motor for rumble effects. It can also be used for vibration actuators that only vibrate part of the gamepad (eg, a single trigger or button). A "dual-rumble" actuator represents the typical low-frequency/high-frequency rumble configuration found on most modern gamepads._

> _In the initial version of the API, only dual-rumble effects are supported. It may be desirable to allow dual-rumble effects to be played back on a "vibration"-type actuator, but how this might be accomplished is outside the scope of this specification._

Does it make sense?

Reviewer: I updated the suggestion. Please take a look.

Author: Perfect!

- _params_

  - : An object to describe a desired haptic effect.

    Expected values are:

    - `duration`: The duration of the effect in milliseconds.
    - `startDelay`: The delay in milliseconds before the effect is started.
    - `strongMagnitude`: Rumble intensity of the low-frequency (strong) rumble motors, normalized to the range between 0.0 and 1.0.
    - `weakMagnitude`: Rumble intensity of the high-frequency (weak) rumble motors, normalized to the range between 0.0 and 1.0.

> > **Note:** Repeated calls to `playEffect()` override the previous calls if they are still ongoing.

Reviewer: ```suggestion
> **Note:** A new call to `playEffect()` overrides a previous ongoing call.
```

Reviewer: my issue with the previous wording is that it implies multiples where none are possible. For example, a call can't override 'previous calls' because (if I'm understanding the logic correct) override means there can be only one effect going at a time. If you don 't see what I'm getting at, please ask.

### Return value

> A promise that resolves with a value of `true` when the playEffect has successfully completed.

Reviewer: ```suggestion
A promise that resolves with `true` when the playEffect successfully completes.
```
